### PR TITLE
OverflowTooltip: Don't render component when text is empty

### PR DESCRIPTION
**Pull request by robinjmurphy**
This PR fixes a bug that I believe was introduced in #4372 where `Tooltip` components could be instantiated without a `title` (which is a require prop).

After upgrading Backstage I noticed this error in the console:

Screenshot 2021-02-25 at 12 45 28

I believe this is happening because some of the components in our catalog don't have a `description` (which is an optional property) but we're attempting to render an `OverflowTooltip`:

https://github.com/backstage/backstage/blob/647bbb14f39a0399cee5b33f64291afbe507685e/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx#L95-L98

This leads to us creating a `Tooltip` without a title:

https://github.com/backstage/backstage/blob/647bbb14f39a0399cee5b33f64291afbe507685e/packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx#L44-L48

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes

### CODE REVIEW

**Review by adamdmharvey — COMMENTED** [review]
Good find; wonder if that could use a test where no content is passed to validate it?

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4697-d2lknvtojsehnrycnpfkcpxp6mavfv8m.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037b645fcb22d9617604094

**Comment by robinjmurphy** [comment]
I've added a basic test. It does show the warning before my fix, however, because it's only a PropTypes error it doesn't actually fail the test. Let me know if you've got any ideas around this (I'm new to testing wiht Jest).

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4697-d2lknvtojsehnrycnpfkcpxp6mavfv8m.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037b645fcb22d9617604094

File: packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx [hunk 1]
```diff
@@ -40,6 +40,10 @@ export const OverflowTooltip = (props: Props) => {
     setHover(truncated);
   };
 
+  if (!props.text) {
+    return null;
+  }
+
   return (
     <Tooltip
       title={props.title ?? props.text!}
```
Comment (benjdlambert) [inline packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx:49]: Think we should be able to remove the `!` from `props.text` now too, and then it's good to go for me!

File: packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx [hunk 2]
```diff
@@ -40,6 +40,10 @@ export const OverflowTooltip = (props: Props) => {
     setHover(truncated);
   };
 
+  if (!props.text) {
+    return null;
```
Comment (Rugvip) [inline packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx:44]: Tbh I'm wondering if this is actually what you'd expect here. The `TextTruncate` component ends up defaulting `text` to `''` if it's not set, perhaps we should change the `Tooltip` `title` to do the same instead? I'm thinking you might get a bit of UI flicker if switching between `null` and text compared to `''` and text.

**Review by Rugvip — COMMENTED** [review]
Thanks!

**Review by Rugvip — APPROVED** [review]
Thanks!

**Pull request merged**